bin/facades.php

```diff
@@ -185,70 +192,221 @@ function resolveDocParamType($method, $parameter)
+        return '('.collect($typeNode->types)
+            ->map(fn ($node) => resolveDocblockTypes($method, $node))
+            ->unique()
+            ->implode('&').')';
+    }
 
-    $types = Str::of($types)
-        ->replace(['$this', 'static'], '\\'.$method->sourceClass()->getName())
-        ->replace('self', '\\'.$method->getDeclaringClass()->getName())
-        ->toString();
+    if ($typeNode instanceof GenericTypeNode) {
+        return resolveDocblockTypes($method, $typeNode->type);
+    }
+
+    if ($typeNode instanceof ThisTypeNode) {
+        return '\\'.$method->sourceClass()->getName();
+    }
+
+    if ($typeNode instanceof ArrayTypeNode) {
+        return 'array';
+    }
+
+    if ($typeNode instanceof IdentifierTypeNode) {
+        if ($typeNode->name === 'static') {
+            return '\\'.$method->sourceClass()->getName();
+        }
+
+        if ($typeNode->name === 'self') {
+            return '\\'.$method->getDeclaringClass()->getName();
+        }
+
+        if (isBuiltIn($typeNode->name)) {
+            return (string) $typeNode;
+        }
+
+        if (class_exists($typeNode->name)) {
+            return (string) $typeNode;
+        }
+
+        if (interface_exists($typeNode->name)) {
+            return (string) $typeNode;
+        }
+
+        if (enum_exists($typeNode->name)) {
+            return (string) $typeNode;
+        }
+
+        if (isKnownOptionalDependency($typeNode->name)) {
+            return (string) $typeNode;
+        }
+
+        if ($typeNode->name === 'class-string') {
+            return 'string';
+        }
+
+        return handleUnknownIdentifierType($method, $typeNode);
+    }
 
-    return stripGenerics($method, $types);
+    if ($typeNode instanceof ConditionalTypeNode) {
+        return handleConditionalType($method, $typeNode);
+    }
+
+    if ($typeNode instanceof NullableTypeNode) {
+        return '?'.resolveDocblockTypes($method, $typeNode->type);
+    }
+
+    if ($typeNode instanceof CallableTypeNode) {
+        return resolveDocblockTypes($method, $typeNode->identifier);
+    }
+
+    echo 'Unhandled type: '.$typeNode::class;
+    echo PHP_EOL;
+    echo 'You will need to update the `resolveDocblockTypes` to handle this type.';
+    exit(1);
 }
 
 /**
- * Remove generics from the type.
+ * Handle conditional types.
  *
- * Unfortunately the @template tag is not currently working with the @method
- * docblocks, so we are stripping them out.
+ * @param  \ReflectionMethodDecorator  $method
+ * @param  \PHPStan\PhpDocParser\Ast\Type\ConditionalTypeNode  $typeNode
+ * @return string
+ */
+function handleConditionalType($method, $typeNode)
+{
+    if (
+        in_array($method->getname(), ['pull', 'get']) &&
+        $method->getDeclaringClass()->getName() === Repository::class
+    ) {
+        return 'mixed';
+    }
+
+    echo 'Found unknown conditional type. You will need to update the `handleConditionalType` to handle this new conditional type.';
+    exit(1);
```

Comment:
This will now error when encountering conditional return types, which means we won't auto-commit wrong values.

I don't think we can assume they should all default to `mixed` yet.